src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,204 @@
+                String task = memo.substring(0, 1);
+
+                // Retrieve task status
+                int indexOfFirstBreak = memo.indexOf("|");
+                String status = memo.substring(indexOfFirstBreak + 2, indexOfFirstBreak + 3);
+                boolean statusIsDone = status.equals("1");
+
+                // skip "| x | " to get task description
+                String descriptionAndTime = memo.substring(indexOfFirstBreak + 6);
+                int indexOfThirdBreak = descriptionAndTime.indexOf("|");
+                String description = descriptionAndTime;
+                String time = "";
+
+                // if time exists, retrieve time and update task description
+                if (indexOfThirdBreak != -1) {
+                    description = descriptionAndTime.substring(0, indexOfThirdBreak - 1);
+                    time = descriptionAndTime.substring(indexOfThirdBreak + 2);
+                }
+
+                // update list
+                Task currentTask = createTask(task, description, time, statusIsDone);
+                pastTasks.add(currentTask);
+            }
+            s.close();
+        } catch (FileNotFoundException e) {
+            System.out.println("     " + e.getMessage());
+        }
+        return pastTasks;
+    }
+
+    /**
+     * Creates a Task object using information retrieved from memory.
+     *
+     * @param task Task identifier.
+     * @param description Task description.
+     * @param time Time associated with Task.
+     * @param isDone Status of Task.
+     * @return Task object.
+     */
+    public Task createTask(String task, String description, String time, boolean isDone) {
+        Task newTask = null;
+        LocalDate date = null;
+        if (time != "") {
+            date = LocalDate.parse(time);
+        }
+        switch (task) {
+        case "T":
+            newTask = new ToDo(description);
+            break;
+        case "D":
+            newTask = new Deadline(description, date);
+            break;
+        case "E":
+            newTask = new Event(description, date);
+            break;
+        default:
+            throw new DukeException("OOPS!!! The Disk memory is invalid");
+        }
+        newTask.setTaskStatus(isDone);
+        return newTask;
+    }
+
+    /**
+     * Stores Task information onto memory.
+     *
+     * @param task Task information.
+     * @throws IOException if unable to write to file.
+     */
+    public void addTaskToDisk(String task) throws IOException {
+        appendToFile(filePath, task);
+    }
+
+    /**
+     * Rewrites the .txt file that contains all the tasks added in order to reflect changes in a task status.
+     *
+     * @param taskNumber the row of Task which changed status.
+     * @param isDone the new status of the task.
+     * @throws IOException if unable to write to file.
+     */
+    public void setTaskStatusOnDisk(int taskNumber, boolean isDone)throws IOException {
```

Comment:
Slight nit, add a space between the parameters and throws.
```suggestion
public void setTaskStatusOnDisk(int taskNumber, boolean isDone) throws IOException {
```